source/common/network/udp_listener_impl.cc

```diff
@@ -31,31 +31,26 @@ namespace Network {
 UdpListenerImpl::UdpListenerImpl(Event::DispatcherImpl& dispatcher, SocketSharedPtr socket,
                                  UdpListenerCallbacks& cb, TimeSource& time_source)
     : BaseListenerImpl(dispatcher, std::move(socket)), cb_(cb), time_source_(time_source) {
-  file_event_ = socket_->ioHandle().createFileEvent(
+  socket_->ioHandle().initializeFileEvent(
       dispatcher, [this](uint32_t events) -> void { onSocketEvent(events); },
       Event::PlatformDefaultTriggerType, Event::FileReadyType::Read | Event::FileReadyType::Write);
 
-  ASSERT(file_event_);
-
   if (!Network::Socket::applyOptions(socket_->options(), *socket_,
                                      envoy::config::core::v3::SocketOption::STATE_BOUND)) {
     throw CreateListenerException(fmt::format("cannot set post-bound socket option on socket: {}",
                                               socket_->localAddress()->asString()));
   }
 }
 
-UdpListenerImpl::~UdpListenerImpl() {
-  disableEvent();
-  file_event_.reset();
-}
+UdpListenerImpl::~UdpListenerImpl() { socket_->ioHandle().resetFileEvents(); }
```

Comment:
I did not want to close the socket here because this would be a difference in behavior. As long as the socket exists and it the `ioHandle` is not a dangling reference then I think this should be good. Is there such a case?

Reply:
Whatever we do here I guess we should do for the tcp listener as well?

Reply:
I take it back, I guess they're different because of shared fds